src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1008,9 +1010,26 @@ private TokenValidationResult ValidateToken(string token, JsonWebToken outerToke
                 }
                 catch (Exception ex)
                 {
-                    // Keep going with the validation as the TokenValidationParameters may have the issuer and signing key set
+                    // The exception is not re-thrown here as the TokenValidationParameters may have the issuer and signing key set
                     // directly on them.
                     LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParameters.ConfigurationManager.MetadataAddress, ex.ToString()));
+
+                    // Retry configuration retrieval on network failure (ServiceUnavailable or RequestTimeout).
+                    if (ex.GetType().Equals(typeof(InvalidOperationException)) && ex.Message.Contains("IDX20803 :")
```

Comment:
>ex.Message.Contains("IDX20803 :") [](http://example.com/codeflow?start=82&length=33)

This is not needed:
1. The GetBaseConfigurationAsync() only throws IOException with "IDX20803 :".
2. If GetBaseConfigurationAsync() changes the message this may break.

Reply:
That's a good point; removing this makes the code more resilient to future changes in log messages. Fixed. :) https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1784/commits/47e0c5bf3236c4fc08d1f0f56dd250d5e6051ddb

Reply:
Good point! Addressed :) https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1784/commits/47e0c5bf3236c4fc08d1f0f56dd250d5e6051ddb